### PR TITLE
Sign up: Add updated Site Intent step (WIP)

### LINKED ISSUE #90628: Task: Create signup intent capture step in Stepper
### Details

Build a tentative step to collect user intent in Stepper. It should look something like this:

image

image

We already have `IntentStep`. We should probably extend that.

When the user clicks, the steps should submit the intention to the flow. 

_Note: this design is not final._ 

Design is here: https://www.figma.com/design/3u71fUJraClkRQiLrAORPf/Trail-Map-Project?node-id=3906-5213&t=C4ETPSlpGWgigLG6-0

### Checklist

_No response_

### Related

_No response_

### PR DESCRIPTION
**Pull request by agrullon95**
Related to https://github.com/Automattic/wp-calypso/issues/90628

## Proposed Changes

* Add updated site intent screen based on designs from: https://www.figma.com/design/3u71fUJraClkRQiLrAORPf/Trail-Map-Project?node-id=3908-7311&m=dev

This will use the following segmentation survey API changes to retrieve the correct questions/information: **D148812**

Designs have not been finalized yet.

## Why are these changes being made?

* These changes are part of the initiative to improve the sign-up flow by helping tailor the onboarding experience based on the selected site intent option.

## Testing Instructions

* Checkout branch or use live link
* Checkout / patch this diff: D148812
* Navigate to `/setup/site-setup/intent?siteSlug={SITE_SLUG}`
* Compare changes to design (designs have not been finalized yet).

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/site-setup-flow.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ const siteSetupFlow: Flow = {
 	useSteps() {
 		return [
 			STEPS.GOALS,
-			STEPS.INTENT,
+			STEPS.INTENT_I2,
```
Comment (agrullon95) [inline client/landing/stepper/declarative-flow/site-setup-flow.ts:68]: Temporarily changed to easily show changes in this PR within the `/site-setup` flow.

File: client/landing/stepper/declarative-flow/internals/steps-repository/intent-step-i2/intents.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Onboard } from '@automattic/data-stores';
+import { useTranslate } from 'i18n-calypso';
+import { Intent } from './types';
+
+export const useIntents = (): Intent[] => {
+	const translate = useTranslate();
+	const SiteIntent = Onboard.SiteIntent;
+
+	const intents: Intent[] = [
```
Comment (agrullon95) [inline client/landing/stepper/declarative-flow/internals/steps-repository/intent-step-i2/intents.tsx]: Work in progress. Intents information will be updated when its finalized.

**Review by alshakero — CHANGES_REQUESTED** [review]
Beautiful! Can we give the cards some hover state?

**Comment by agrullon95** [comment]
Closing this PR and moving changes to: https://github.com/Automattic/wp-calypso/pull/90788

**Pull request closed without merging**